Mathlib/Dynamics/FixedPoints/Increasing.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2024 Shuhao Song. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shuhao Song, Yury Kudryashov
+-/
+import Mathlib.Order.Filter.EventuallyConst
+import Mathlib.Order.Iterate
+import Mathlib.Order.WellFounded
+import Mathlib.Data.Fintype.Card
+
+/-!
+# Fixed points of function `f` with `f(x) ≥ x`
+
+In this file we consider the fixed points of function `f : α → α` with `f(x) ≥ x`. If `α` is
+a finite type, then the sequence `x, f(x), f(f(x)), ...` will eventually be a constant.
+
+# Main definitions
+* `fixedPointIndex`: The smallest `n` such that the sequence `x, f(x), f(f(x)), ...`
+  becomes a constant after this index.
+* `eventuallyValue`: The value that the sequence converges to after sufficient number
+  of iterations of `f` on `x`.
+-/
+namespace Function
+
+open Filter
+variable {α : Type*} [PartialOrder α] [Finite α] {f : α → α}
```

Comment:
I don't think any part of the file actually requires finitude, only the well-foundedness condition. Note that we have [`Finite.wellFounded_of_trans_of_irrefl`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Fintype/Card.html#Finite.wellFounded_of_trans_of_irrefl).

Reply:
We have `Finite.to_wellFoundedGT`.

Reply:
That's even better.